src/zenml/zen_stores/schemas/pipeline_deployment_schemas.py

```diff
@@ -121,21 +130,18 @@ class PipelineDeploymentSchema(BaseSchema, table=True):
         ondelete="SET NULL",
         nullable=True,
     )
+
+    # SQLModel Relationships
+    user: Optional["UserSchema"] = Relationship()
+    workspace: "WorkspaceSchema" = Relationship()
+    stack: "StackSchema" = Relationship()
+    pipeline: "PipelineSchema" = Relationship()
+    schedule: Optional["ScheduleSchema"] = Relationship()
+    build: Optional["PipelineBuildSchema"] = Relationship()
     code_reference: Optional["CodeReferenceSchema"] = Relationship()
 
-    runs: List["PipelineRunSchema"] = Relationship(back_populates="deployment")
-
-    run_name_template: str
-    pipeline_configuration: str = Field(sa_column=Column(TEXT, nullable=False))
-    step_configurations: str = Field(
-        sa_column=Column(
-            String(length=MEDIUMTEXT_MAX_LENGTH).with_variant(
-                MEDIUMTEXT, "mysql"
-            ),
-            nullable=False,
-        )
-    )
-    client_environment: str = Field(sa_column=Column(TEXT, nullable=False))
+    pipeline_runs: List["PipelineRunSchema"] = Relationship()
+    step_runs: List["StepRunSchema"] = Relationship()
```

Comment:
These should cascade delete I think

Reply:
For both `pipeline_runs` and `step_runs`, the `cascade delete` behavior is already defined when we define the foreign keys:

```python
    deployment_id: UUID = build_foreign_key_field(
        source=__tablename__,
        target=PipelineDeploymentSchema.__tablename__,
        source_column="deployment_id",
        target_column="id",
        ondelete="CASCADE",
        nullable=True,
    )
```

Defining these args in the `Relationship` object only matters if you do the deletion (or any other operation for that matter) on the schema object directly which we never do. I think defining such relationships creates a lot of confusion in the code that's why I was trying to remove the redundant parts as much as possible.